honeybadger/protocols.py

```diff
@@ -0,0 +1,30 @@
+from typing import Protocol, Dict, Any, Optional, List
+from .types import EventsSendResult, Notice, Event
+
+
+class Connection(Protocol):
```

Comment:
In PHP, we have a similar interface called `Reporter` (note: I'm not the one that came up with it). Just mentioning in case you like it more than the current name 😄).

Reply:
I like the intent of "Reporter", but since this is a protocol for connections, I think it makes it easier to know which classes we are linking to if they are similarly named.